Makefile

```diff
@@ -1,17 +1,46 @@
-.PHONY: build run default
+.PHONY: dg doc build run test clean
 
-IMAGE_NAME = dovecot
+DISTRO = fedora-26-x86_64
+VARIANT = upstream
+DG = /usr/bin/dg
+GOMD2MAN = /usr/bin/go-md2man
+DOCKERFILE_SRC := Dockerfile.template
+DOCKERFILE := Dockerfile
+TEST_IMAGE_NAME := container-images-tests
 
+SELECTORS = --distro ${DISTRO}.yaml --multispec-selector variant=${VARIANT}
+DG_EXEC = ${DG} --max-passes 25 --spec specs/configuration.yml --multispec specs/multispec.yml ${SELECTORS}
+DISTRO_ID = $(shell ${DG_EXEC} --template "{{ config.os.id }}")
+IMAGE_REPOSITORY = $(shell ${DG_EXEC} --template "{{ spec.image_repository }}")
 
-default: run
+install-dependencies:
+	./requirements.sh
 
-build:
-	docker build --tag=$(IMAGE_NAME) .
+dg:
+	${DG_EXEC} --template $(DOCKERFILE_SRC) --output $(DOCKERFILE)
+	${DG_EXEC} --template help/help.md --output help/help.md.rendered
+
+doc: dg
+	mkdir -p ./root/
+	${GOMD2MAN} -in=help/help.md.rendered -out=./root/help.1
+
+build: doc dg
+	sudo chown -R $(shell id -u) .
+	docker build --tag=${IMAGE_REPOSITORY} -f $(DOCKERFILE) .
 
 run: build
-	docker run -v /var/spool/postfix:/var/spool/postfix \
-	           -v /var/spool/mail:/var/spool/mail \
-	           -e MYHOSTNAME=localhost $(IMAGE_NAME)
+	docker run -e MYHOSTNAME=localhost $(IMAGE_REPOSITORY)
+
+test: build
+	cd tests; MODULE=docker URL="docker=${IMAGE_REPOSITORY}" DOCKERFILE="../$(DOCKERFILE)" VERSION=${VERSION} DISTRO=${DISTRO} mtf -l *.py
+
+test-in-container: test-image
+	docker run --rm -ti -v /var/run/docker.sock:/var/run/docker.sock:Z -v ${PWD}:/src ${TEST_IMAGE_NAME} "SELECTORS=${SELECTORS}"
+
+test-image:
+	docker build --tag=${TEST_IMAGE_NAME} -f ./Dockerfile.tests .
```

Comment:
It seems that there's no `Dockerfile.tests` in this repo, so this target seems to be useless. I'd suggest removing it.

Reply:
+ also the `test-in-container` target.